patches/tModLoader/Terraria/ModLoader/ModBiome.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Generic;
+using Terraria.WorldBuilding;
+
+namespace Terraria.ModLoader
+{
+	/// <summary>
+	/// This class represents a biome added by a mod. It exists to centralize various biome related hooks, handling a lot of biome boilerplate. Use its various logic hooks act as if they were in ModPlayer, and use ModifyWorldGenTasks as if it were in ModWorld.
+	/// </summary>
+	public abstract class ModBiome : ModType
+	{
+		internal int index;
+
+		protected override void Register() {
+			ModTypeLookup<ModBiome>.Register(this);
+			BiomeLoader.Add(this);
+		}
+
+		internal void Update(Player player, ref bool value) {
```

Comment:
I think it makes more sense to move this method into the loop of `BiomeLoader.Update`